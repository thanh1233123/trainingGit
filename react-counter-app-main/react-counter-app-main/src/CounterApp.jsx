import PropTypes from 'prop-types';
import { useState } from 'react';


export const CounterApp = ({ value }) => {
  const [ counter, setCounter ] = useState( value );

  const handleAdd = () => setCounter( counter + 1 );
  const handleSubstract = () => setCounter( counter - 1 );
  const handleReset = () => setCounter( value );
  
  return (
    <>
      <h1>CounterApp</h1>
      <h2> { counter } </h2>

      <button onClick={ handleAdd } data-testid="btn-add"> +1 </button>
      <button onClick={ handleSubstract } data-testid="btn-substract"> -1 </button>
      <button onClick={ handleReset } data-testid="btn-reset"> Reset </button>
    </>
  );
}

CounterApp.propTypes = {
  value: PropTypes.number.isRequired,
}