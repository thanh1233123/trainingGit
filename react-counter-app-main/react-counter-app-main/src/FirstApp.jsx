import PropTypes from 'prop-types';

const newMessage = [1, 2, 3, 4, 5, 6, 7, 8, 9];

const getResult = (a, b) => {
  return a + b;
}

// Promise object => render error
// const getResultAsync = async (a, b) => {
//   return a + b;
// }

export const FirstApp = ({ title, subTitle, version }) => {
  // console.log(props);

  return (
    <>
      <h1 data-testid="test-title">{ title }</h1>
      <h2>{ subTitle }</h2>
      <p>{ `Version: ${version}` }</p>
      <p>{ newMessage }</p>
      <h2>{ subTitle }</h2>
      <p>{ getResult(7, 2) }</p>
      {/* <p>{ getResultAsync(7, 2) }</p> */}
    </>
  )
}

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subTitle: PropTypes.string,
  version: PropTypes.number
}

FirstApp.defaultProps = {
  subTitle: 'There is no subtitle',
  version: 1.0
}