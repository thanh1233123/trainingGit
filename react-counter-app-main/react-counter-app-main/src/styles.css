html, body {
  background-color: #21232A;
  color: white;
  font-family: Helvetica, Arial, sans-serif;
  font-size: 1.3rem;
  padding: 1rem;
}

button {
  padding: 15px;
  font-size: 1.3rem;
  margin-right: 6px;
  cursor: pointer;
}