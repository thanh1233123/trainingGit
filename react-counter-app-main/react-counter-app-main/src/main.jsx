import React from 'react';
import ReactDOM from 'react-dom/client';
import './styles.css';
import { CounterApp } from './CounterApp';


ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    <CounterApp value={ 10 } />
  </React.StrictMode>
);