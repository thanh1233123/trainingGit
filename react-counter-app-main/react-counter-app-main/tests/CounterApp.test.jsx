import { describe, test, it, expect } from 'vitest';
import { screen, render, fireEvent } from '@testing-library/react';
import { CounterApp } from '../src/CounterApp';

describe('Tests in <CounterApp />', () => {
  const initialValue = 100;

  it('should match with snapshot', () => {
    const { container } = render( <CounterApp value={ initialValue } /> );
    expect( container ).toMatchSnapshot();
  });

  test('initial value should be 100', () => { 
    render( <CounterApp value={ initialValue } /> );
    expect( screen.getByText(initialValue) ).toBeTruthy();
  });

  it('should increment with +1 button', () => { 
    render( <CounterApp value={ initialValue } /> );
    fireEvent.click( screen.getByTestId('btn-add') );
    expect( screen.getByText(initialValue + 1)).toBeTruthy();
  });

  it('should decrement with -1 button', () => { 
    render( <CounterApp value={ initialValue } /> );
    fireEvent.click( screen.getByTestId('btn-substract') );
    expect( screen.getByText(initialValue - 1)).toBeTruthy();
  });

  it('should reset with reset button', () => { 
    render( <CounterApp value={ initialValue } /> );
    fireEvent.click( screen.getByTestId('btn-add') );
    fireEvent.click( screen.getByTestId('btn-add') );
    fireEvent.click( screen.getByTestId('btn-add') );
    fireEvent.click( screen.getByTestId('btn-reset') );
    expect( screen.getByText(initialValue) ).toBeTruthy();
  });
});
