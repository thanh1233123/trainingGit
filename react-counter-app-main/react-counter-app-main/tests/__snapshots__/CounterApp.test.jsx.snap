// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Tests in <CounterApp /> > should match with snapshot 1`] = `
<div>
  <h1>
    CounterApp
  </h1>
  <h2>
     
    100
     
  </h2>
  <button
    data-testid="btn-add"
  >
     +1 
  </button>
  <button
    data-testid="btn-substract"
  >
     -1 
  </button>
  <button
    data-testid="btn-reset"
  >
     Reset 
  </button>
</div>
`;
